python/pyspark/ml/base.py

```diff
@@ -18,13 +18,40 @@
 from abc import ABCMeta, abstractmethod
 
 import copy
+import threading
 
 from pyspark import since
-from pyspark.ml.param import Params
 from pyspark.ml.param.shared import *
 from pyspark.ml.common import inherit_doc
 from pyspark.sql.functions import udf
-from pyspark.sql.types import StructField, StructType, DoubleType
+from pyspark.sql.types import StructField, StructType
+
+
+class FitMutlipleIterator(object):
```

Comment:
What about change this `FitMutlipleIterator` class to be an inner class in default implementation method `fitMultiple` ? I think put it outside will be no other usage.

Reply:
I'm open to this, but I didn't initially do it this way because I've been bit by nested classes in python before. There are subtle issues with nested classes in python. The one that comes to mind is serialization (which isn't an issue here) but that's not the only one.

Reply:
@jkbradley @WeichenXu123 I made `FitMultipleIterator` a private class, is that good enough or should I make it internal to the `fitMultiple` method?